### DISCUSSION TITLE
LUT questions (HDR threshold, restart HyperHDR)

### DISCUSSION THREAD

**Discussion by robinkoehler** [awawa-dev/HyperHDR#124] [Q&A] [answered]
Hi Awawa and everyone!

I'm enjoying the v17 official update, and am having another go at creating LUTs. For SDR, I have the settings worked out. For HDR however, I struggle to set the Min. threshold values for each colour correctly. Even extremely low values (1-3) eliminate darker colours that would require a value of 30-50 to achieve the same effect in SDR. The HDR min threshold seems to be very aggressive. 

I have read the changelog and found "fix values premature clipping in the LUT generator" and am not sure if this is related to this issue. I have deleted the browser cache and reloaded the page to make sure I was not accidentally loading a previous version of the LUT generator. 

Secondly, I was wondering what the correct and easy way to restart HyperHDR is after replacing the .3d file with my updated version? I'm running it on Raspberry Pi and have only been able to get the new LUT loaded in by rebooting the Raspberry Pi. Is there a command to load the new LUT or another way to restart the program to avoid a full reboot?

Thank you kindly in advance!

Best wishes, 

Robin

**Comment by awawa-dev** [comment, marked as answer]
Hi Robin
> I'm enjoying the v17 official update, and am having another go at creating LUTs. For SDR, I have the settings worked out. For HDR however, I struggle to set the Min. threshold values for each colour correctly. Even extremely low values (1-3) eliminate darker colours that would require a value of 30-50 to achieve the same effect in SDR. The HDR min threshold seems to be very aggressive.

Yes, HDR processing is very aggressive on its own. For low values like RGB(3,3,3) it brings them to zero RGB(0,0,0) without any truncation so you don't need to truncate them further. On the other hand it bumps higher values quite aggressively so it can't be compared to SDR processing because it leads to some misunderstandings. For example black border detection is working on different limit values in SDR and HDR modes:  ````https://github.com/awawa-dev/HyperHDR/issues/116````  
  
> Secondly, I was wondering what the correct and easy way to restart HyperHDR is after replacing the .3d file with my updated version? I'm running it on Raspberry Pi and have only been able to get the new LUT loaded in by rebooting the Raspberry Pi. Is there a command to load the new LUT or another way to restart the program to avoid a full reboot?

You can restart HyperHDR service using commands from HyperHDR's instalation wiki or just disable and enable again HDR tone mapping in the 'remote' tab: it will reload the LUT table from the file (you can also use JSON API to do that). Verify logs just to be sure.

Best regards
Awawa

**Reply by robinkoehler** [reply]
That makes perfect sense! Just wanted to make sure I'm not doing it wrong.

Enabling/Disabling HDR tone mapping speeds up the process of comparing LUTs immensely as they can just be in the same folder and renamed as needed. Brilliant!

Appreciate this great tool and the instant response :-)
